.github/workflows/security.yml

```
@@ -43,15 +58,22 @@ jobs:
       upload-artifact: true
 
   codeql:
     name: CodeQL Analysis
     runs-on: ubuntu-latest
-    timeout-minutes: 15
-    needs: [build-for-codeql]
+    timeout-minutes: 8  # Reduced from 15min (2x typical run time)
+    needs: [build-for-codeql]  # Removed 'changes' dependency (redundant - already in build-for-codeql)
+    # Skip on ui-only PRs (weekly schedule still runs for comprehensive check)
+    if: |
+      github.event_name == 'schedule' ||
```

actionlint (1.7.8)

property "changes" is not defined in object type {build-for-codeql: {outputs: {artifact-name: string}; result: string}}

(expression)